Include/internal/pycore_global_objects.h

```diff
@@ -66,9 +66,6 @@ struct _Py_static_objects {
 struct _Py_interp_cached_objects {
     PyObject *interned_strings;
 
-    /* AST */
-    PyObject *str_replace_inf;
-
```

Comment:
@Yhg1s - let me know what you think should be done with this PR. It removes `str_replace_inf` from a struct inside `PyInterpreterState`, which triggers the ABI check.

* Keep the field (as a dead field) in 3.13?
* Remove the field (changing the ABI of `PyInterpreterState`)?
* Don't backport this PR (it's a bugfix, but not critical)?

Reply:
Let's just keep it as an unused field. It makes life (a lot!) easier for debuggers if we don't move things around, even if it's not technically part of the public ABI.

Reply:
Oh, I forgot that adding a `_Py_DECLARE_STR()` also changes the ABI. Keeping the field isn't enough.

Reply:
Sigh... It annoys me a little that all this global data is in a flattened struct, such that any change in size of any of the struct completely changes the layout of the structs, and that that means any uses of `_Py_STR` outside the core are impossible to support. Not that I think there are any (yet) (I hope!) but details like that have a tendency to leak out of the interpreter, and it would've been nice to think about this in the design.

Anyway, this is the best we can do, so it'll have to do.